plugin/src/main/groovy/com/btkelly/gnag/utils/ProjectHelper.groovy

```diff
@@ -20,18 +20,31 @@ import org.gradle.api.Project
 /**
  * Created by bobbake4 on 4/19/16.
  */
-class ReportHelper {
+class ProjectHelper {
 
-    private final Project project;
+    private final Project project
 
-    ReportHelper(Project project) {
+    ProjectHelper(Project project) {
         this.project = project
     }
 
-    // todo: this doesn't seem like it belongs in the ReportHelper. refactor?
-    public List<File> getAndroidSources() {
-        project.android.sourceSets.inject([]) {
-            dirs, sourceSet -> dirs + sourceSet.java.srcDirs
+    public Project getProject() {
+        return project
+    }
+
+    public boolean isAndroidProject() {
+        return project.hasProperty("android")
```

Comment:
I agree that would be more robust, let me play with that because I'm not sure if that would work with a pure Java project. My concern would be that `com.android.build.gradle.BasePlugin` doesn't exist on plain old Java projects but it's worth testing.